tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/unfuse_batch_norm.cc

```diff
@@ -200,26 +196,22 @@ Value createReduce(Location loc, Value operand, Value zero,
 }
 
 // Calculate total reduce size, assuming it is a dynamic shape with static rank.
-// Reduce from operand to operand[feature_index]
-Value calculateReduceSize(Operation* op, Value operand,
-                          RankedTensorType operandType,
+// Reduce from operand to operand[feature_index]/scale
+Value calculateReduceSize(Operation *op, Value operand,
+                          RankedTensorType operandType, Value scale,
                           RankedTensorType scaleType, int64_t featureIndex,
-                          PatternRewriter& rewriter) {
+                          PatternRewriter &rewriter) {
   Location loc = op->getLoc();
```

Comment:
Lets use ImplicitLocOpBuilder

```c++
ImplicitLocOpBuilder b(op->getLoc(), rewriter);
```

and just do `b.create<shape::NumElementsOp>(indexType, operandShape);` below. That would IMHO make the structure clearer below.

Reply:
Done